org.jacoco.core/src/org/jacoco/core/internal/runtime/ClassInjectionRuntime.java

```diff
@@ -0,0 +1,297 @@
+		return Class.class //
+				.getMethod("getModule") //
+				.invoke(cls);
+	}
+
+	/**
+	 * Provides access to classes {@code java.lang.invoke.MethodHandles} and
+	 * {@code java.lang.invoke.MethodHandles.Lookup} introduced in Java 8.
+	 */
+	private static class Lookup {
+
+		private final Object instance;
+
+		private Lookup(final Object instance) {
+			this.instance = instance;
+		}
+
+		/**
+		 * @return a lookup object for the caller of this method
+		 */
+		static Lookup lookup() throws Exception {
+			return new Lookup(Class //
+					.forName("java.lang.invoke.MethodHandles") //
+					.getMethod("lookup") //
+					.invoke(null));
+		}
+
+		/**
+		 * @param obj
+		 *            the object from which the method is accessed
+		 * @param name
+		 *            the name of the method
+		 * @param methodType
+		 *            the type of the method
+		 * @return a method handle
+		 */
+		MethodHandle bind(final Object obj, final String name,
+				final MethodType methodType) throws Exception {
+			return new MethodHandle(Class //
+					.forName("java.lang.invoke.MethodHandles$Lookup") //
+					.getMethod("bind", Object.class, String.class,
+							Class.forName("java.lang.invoke.MethodType")) //
+					.invoke(this.instance, obj, name, methodType.instance));
+		}
+
+		/**
+		 * @param refc
+		 *            the class or interface from which the method is accessed
+		 * @param name
+		 *            the field's name
+		 * @param type
+		 *            the field's type
+		 * @return a method handle which can store values into the field
+		 */
+		MethodHandle findStaticSetter(final Class<?> refc, final String name,
+				final Class<?> type) throws Exception {
+			return new MethodHandle(Class //
+					.forName("java.lang.invoke.MethodHandles$Lookup") //
+					.getMethod("findStaticSetter", Class.class, String.class,
+							Class.class) //
+					.invoke(this.instance, refc, name, type));
+		}
+
+		/**
+		 * See corresponding method introduced in Java 9.
+		 *
+		 * @param targetClass
+		 *            the target class
+		 * @param lookup
+		 *            the caller lookup object
+		 * @return a lookup object for the target class, with private access
+		 */
+		static Lookup privateLookupIn(final Class<?> targetClass,
+				final Lookup lookup) throws Exception {
+			return new Lookup(Class //
+					.forName("java.lang.invoke.MethodHandles") //
+					.getMethod("privateLookupIn", Class.class,
+							Class.forName(
+									"java.lang.invoke.MethodHandles$Lookup")) //
+					.invoke(null, targetClass, lookup.instance));
+		}
+
+		/**
+		 * See corresponding method introduced in Java 9.
+		 *
+		 * @param bytes
+		 *            the class bytes
+		 * @return class
+		 */
+		Class<?> defineClass(final byte[] bytes) throws Exception {
+			return (Class<?>) Class //
+					.forName("java.lang.invoke.MethodHandles$Lookup")
+					.getMethod("defineClass", byte[].class)
+					.invoke(this.instance, (Object) bytes);
```

Comment:
@Godin When editing this file Eclipse removed the `(Object)` cast automatically. Oops.

Not sure whether the primitive array will be autoboxed here or if will be a single parameter even without the cast. In the first case a local variable of type Object should be safe.

Reply:
> Not sure whether the primitive array will be autoboxed here or if will be a single parameter even without the cast.

behavior is well defined - `byte[] bytes` is passed as `new Object[] { bytes }`, however confusing as some might wrongly think that each element will be boxed 😉 

> Eclipse removed the (Object) cast automatically

and exactly because of this confusion IntelliJ proposes to do opposite to Eclipse and add explicit cast to `Object` 😆

Reply:
Damn. Will a local variable be a fix point for both IDEs? 🤔 

			final Object arg = bytes;
			return (Class<?>) Class //
					.forName("java.lang.invoke.MethodHandles$Lookup")
					.getMethod("defineClass", byte[].class)
					.invoke(this.instance, arg);

Reply:
Fix point is

```
.invoke(this.instance, new Object[] { bytes });
```